contexts/manifestAppProviders.tsx

```diff
@@ -76,32 +77,49 @@ const ManifestChainProvider = ({ children }: { children: ReactNode }) => {
     },
   };
 
+  // Function to force complete ChainProvider reset
+  const forceChainProviderReset = useCallback(() => {
+    setChainProviderKey(prev => prev + 1);
+  }, []);
+
+  // Expose the reset function to Web3AuthContext
+  const contextValue = useMemo(
+    () => ({
+      ...web3auth,
+      forceChainProviderReset,
+    }),
+    [web3auth, forceChainProviderReset]
+  );
+
   return (
-    <ChainProvider
-      chains={[manifestChain, osmosisChain, axelarChain]}
-      assetLists={[manifestAssets, osmosisAssets, axelarAssets]}
-      wallets={combinedWallets}
-      logLevel={env.production ? 'NONE' : 'INFO'}
-      endpointOptions={endpointOptions}
-      sessionOptions={{ duration: 60 * 60 * 24 * 7 * 1000 }} // 7 days in ms
-      walletConnectOptions={{
-        signClient: {
-          projectId: env.walletConnectKey,
-          relayUrl: 'wss://relay.walletconnect.org',
-          metadata: {
-            name: 'Alberto',
-            description: 'Manifest Network Web App',
-            url: 'https://alberto.com',
-            icons: [],
+    <Web3AuthContext.Provider value={contextValue}>
+      <ChainProvider
+        key={chainProviderKey} // This forces complete re-mount and reset of cosmos-kit
```

Comment:
Cute. Internal React mechanism. The `key` parameter doesn't belong to `ChainProvider`.